pkg/cmd/cli/cmd/rollback.go

```diff
@@ -0,0 +1,85 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kubecmd "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd"
+
+	latestapi "github.com/openshift/origin/pkg/api/latest"
+	client "github.com/openshift/origin/pkg/cmd/client"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+const rollbackLongDesc = `
+Rolls back a deploymentConfig to match a previous deployment. By default, only the pod
+template of the specified deployment will be included in the rollback.
+
+Examples:
+  $ %[1]s %[2]s --from=deployment-2
+  ...
+
+  $ %[1]s %[2]s --from=deployment-3 --include-strategy=true
+  ...
+
+  $ %[1]s %[2]s --from=deployment-1 --output=json
+  ...
+`
+
+func NewCmdRollback(parentName string, name string, f *Factory, out io.Writer) *cobra.Command {
+	rollback := &deployapi.DeploymentConfigRollback{}
+
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s --from=<deployment>", name),
+		Short: "Rolls back a deploymentConfig to match a previous deployment.",
+		Long:  fmt.Sprintf(rollbackLongDesc, parentName, name),
+		Run: func(cmd *cobra.Command, args []string) {
+			from := kubecmd.GetFlagString(cmd, "from")
+			if len(from) == 0 {
+				usageError(cmd, "The --from argument is required.")
+			}
+
+			output := kubecmd.GetFlagString(cmd, "output")
+
+			osClient, _, err := f.Clients(cmd)
+			checkErr(err)
+
+			namespace := getOriginNamespace(cmd)
+
+			newConfig, err := osClient.DeploymentConfigs(namespace).Rollback(rollback)
+			if err != nil {
+				glog.Fatalf("Error generating deploymentConfig for rollback: %v", err)
+			}
+
+			if output == "json" {
+				json, encodeErr := latestapi.Codec.Encode(newConfig)
+				if encodeErr != nil {
+					glog.Fatalf("Couldn't interpret rollback response: %v", err)
```

Comment:
Mix of styles is wrong.  You're fine on checkErr()

----- Original Message -----
> > +			output := kubecmd.GetFlagString(cmd, "output")
> > +
> > +			osClient, _, err := f.Clients(cmd)
> > +			checkErr(err)
> > +
> > +			namespace := getOriginNamespace(cmd)
> > +
> > +			newConfig, err :=
> > osClient.DeploymentConfigs(namespace).Rollback(rollback)
> > +			if err != nil {
> > +				glog.Fatalf("Error generating deploymentConfig for rollback: %v", err)
> > +			}
> > +
> > +			if output == "json" {
> > +				json, encodeErr := latestapi.Codec.Encode(newConfig)
> > +				if encodeErr != nil {
> > +					glog.Fatalf("Couldn't interpret rollback response: %v", err)
> 
> It's not clear when to use it, and I'm seeing a mix of styles. Changed to use
> checkErr. Docs on that function with guidelines would be helpful, if you can
> articulate them here.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/786/files#r23787790